app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodssingle/EnterClaimController.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.rejectedgoodssingle
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import play.api.mvc.Action
+import play.api.mvc.AnyContent
+import play.api.mvc.Call
+import play.api.mvc.Cookie
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.Forms
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.TaxCode
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.{rejectedgoodssingle => pages}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+@Singleton
+class EnterClaimController @Inject() (
+  val jcc: JourneyControllerComponents,
+  enterClaim: pages.enter_claim
+)(implicit val ec: ExecutionContext, viewConfig: ViewConfig)
+    extends RejectedGoodsSingleJourneyBaseController {
+
+  val key               = "enter-claim.rejected-goods.single"
+  val taxCodeCookieName = "taxCode"
+  val postAction: Call  = routes.EnterClaimController.submit()
+
+  def show(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    journey.getNextNdrcDetailsToClaim match {
+      case Some(ndrcDetails) =>
+        val amountPaid = BigDecimal(ndrcDetails.amount)
+        val form       = Forms.claimAmountForm(ndrcDetails.taxType, amountPaid)
+        Future.successful(
+          Ok(enterClaim(form, TaxCode(ndrcDetails.taxType), amountPaid, postAction))
+            .withCookies(Cookie(taxCodeCookieName, ndrcDetails.taxType))
+        )
```

Comment:
I am not sure that is the correct place to send the user. It could be that they have entered all of the claim amounts, or it could be that the have not selected any tax codes to claim.

Reply:
anyway, we have to handle None here

Reply:
I would leave that check for the claims review page, and if none duty selected it should send the user back to the selection stage

Reply:
I have added 2 cases, if we have completed the reimbursement claims then go to the `total_reimbursements` page, otherwise go the the `select-duties` page.